File: est datos/Js/matrices.js
function CrearMatriz(filas, columnas) {
    let matriz = new Array(filas);
    for (let i = 0; i < filas; i++) {
        matriz[i] = new Array(columnas);
    }
    return matriz;
}


let filas = parseInt(prompt("Ingrese el numero de filas:"));
let columnas = parseInt(prompt("Ingrese el numero de columnas:"));

let matriz = CrearMatriz(filas, columnas);

for (let i = 0; i < filas; i++) {
    for (let j = 0; j < columnas; j++) {

        matriz[i][j] = prompt(`Ingrese el valor para la posicion [${i+1}][${j+1}]:`);
    }

}
do{
let opcion = parseInt(prompt("De que manera quiere ver la matriz? (1: filas y columnas, 2: columnas y filas)"));
if (opcion == 1) {
    console.log("Matriz por filas y columnas:");
    for (let i = 0; i < filas; i++) {
    for (let j = 0; j < columnas; j++) {
        console.log(`matriz[${i+1}][${j+1}] = ${matriz[i][j]}`);
}}
} else if (opcion == 2) {
    console.log("Matriz por columnas y filas:");
    for (let i = 0; i < columnas; i++) {
    for (let j = 0; j < filas; j++) {
        console.log(`matriz[${j+1}][${i+1}] = ${matriz[j][i]}`);
}}
} else {
    console.log("Opcion no valida.");
}
}while (opcion != 1 && opcion != 2);

File: est datos/java/matrices.java
import java.util.Scanner;
public class matrices {
public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    System.out.println("Ingrese el numero de filas:");
    int filas = sc.nextInt();
    System.out.println("Ingrese el numero de columnas:");
    int columnas = sc.nextInt();
    sc.nextLine();

    int[][] matriz = new int[filas][columnas];

    System.out.println("Vamos a llenar la matriz:");

    for (int i = 0 ; i < filas ; i++){
        for (int j = 0 ; j < columnas ; j++){
            System.out.println("Ingrese un valor entero para la posicion [" + (i+1) + "][" + (j+1) + "]:");
            int dato = sc.nextInt();
            matriz[i][j] = dato;
        } 
    }

    int option;

    do {
        System.out.println("De que manera quiere ver la matriz? (1: filas, 2: columnas)");
        option = sc.nextInt();
        
        
        if (option == 1){
            System.out.println("La matriz por filas es:");
            for (int i = 0 ; i < filas ; i++){
                for (int j = 0 ; j < columnas ; j++){
                    System.out.print(matriz[i][j] + " ");
                }
                System.out.println();
            }
         } else if (option == 2){
                System.out.println("La matriz por columnas es:");
                for (int j = 0 ; j < columnas ; j++){
                    for (int i = 0 ; i < filas ; i++){
                        System.out.print(matriz[i][j] + " ");
                    }
                    System.out.println();
                }
            } else {
                System.out.println("Opcion no valida.");
            }

        }while(option != 1 && option != 2);
        System.out.println("Gracias por usar el programa.");
    sc.close();

}
}